demos/immersed_fem/immersed_fem.py.rst

```diff
@@ -90,8 +126,212 @@ you can type the following command in the terminal
+anything after in the list is treated as a hole in the domain. These need
+to be registered to the model with ``syncronize`` before we can use them.
+
+.. code-block:: none
+
+   rectangle_loop = model.geo.addCurveLoop(rectangle_lines, 9)
+   circle_loop = model.geo.addCurveLoop(circle_arcs, 10)
+
+   punched_surface = model.geo.addPlaneSurface([rectangle_loop,circle_loop], 1)
+   circle_surface = model.geo.addPlaneSurface([circle_loop], 2)
+   model.geo.synchronize()
+   
+Finally, we group together some edges and define ``Physical`` entities.
+
+.. code-block:: none
+
+   model.addPhysicalGroup(1, [rectangle_lines[1], rectangle_lines[3]], 11, name = "HorEdges")
+   model.addPhysicalGroup(1, [rectangle_lines[0], rectangle_lines[2]], 12, name = "VerEdges")
+   model.addPhysicalGroup(1, circle_arcs, 13, name = "Circle")
+   model.addPhysicalGroup(2, [punched_surface], 3, name = "PunchedDom")
+   model.addPhysicalGroup(2, [circle_surface], 4, name = "Disc")
+
+A number of meshing options are available. In particular, the algorithm for mesh generation
+and can be set globally or for individual surfaces.
+
+* 2D: 1: MeshAdapt, 2: Automatic, 3: Initial mesh only, 5: Delaunay, 6: Frontal-Delaunay (default),\
+  7: BAMG, 8: Frontal-Delaunay for Quads, 9: Packing of Parallelograms, 11: Quasi-structured Quad
+* 3D: Delaunay (default) 3: Initial mesh only 4: Frontal 7: MMG3D 9: R-tree 10: HXT
+
+For more information see the `Gmesh documentation on algorithms <https://gmsh.info/doc/texinfo/gmsh.html#Choosing-the-right-unstructured-algorithm>`_.
+
+When writing the mesh to file, the format is determined by the file extension. For example,
+`.msh2` for Gmsh 2.x, `.msh` for GMSH 4.x.
+
+.. code-block:: none
+
+   gmsh.option.setNumber("Mesh.Algorithm", 6)
+   gmsh.option.setNumber("Mesh.MshFileVersion", 4.1)
+   gmsh.model.mesh.generate(2)
+   gmsh.write('gmsh_api_demo.msh')
+   gmsh.finalize()
+
+.. note::
+   
+   Best practice is to shut down the Gmsh API kernel after finalising the mesh.
+
+Using OpenCASCADE through Gmsh
+------------------------------
+
+Using OpenCASCADE through Gmsh, we define higher level geometries such as rectangles 
+and discs directly. It also has addition 3D capability not illustrated here.
+
+As with the Gmsh API, we ``initialize`` and start constructing a new mesh model.
+
+.. code-block:: none
+
+   gmsh.initialize()
+   gmsh.model.add("gmsh_occ_demo")
+
+We first use OpenCASCADE to create a rectangle and a cylinder object. This automates
+the create of points, lines, and surfaces. Both objects need to be registered to the 
+model with ``syncronize`` before we can use them.
+
+.. code-block:: none
+
+   rectangle_obj = gmsh.model.occ.addRectangle(x0 - L/2, y0 - W/2, z0, L, W, 1)
+   cylinder_obj = gmsh.model.occ.addCylinder(x0, y0, z0, 0, 0, 1, R, 2)
+   gmsh.model.occ.synchronize()
+
+As the default behavior of the ``cut`` function, utilized to create the hole, is to 
```

Comment:
To create the whole, we use the method `occ.cut` with the option `removeTool=False` to retain the disc interior (which would be otherwise deleted by default).